feat: show password eye button created at register page

sumanjeet0012 opened the pull request:
## Related Issue

<!-- If there is no issue number, the PR will not be merged. Therefore, please ensure that the issue number is added -->
fix: #107

## Description

<!-- Please provide more context or information for us to properly rewrite your statement  -->

Added show password checkbox which can be used to show or hide the password.

## Screenshots

<!-- Add screenshots to preview the changes  -->

![image](https://github.com/Sanchitbajaj02/palettegram/assets/78401409/47dd18b2-b841-46ac-8ca8-6bd8566a148e)

Sanchitbajaj02 commented:
@sumanjeet0012 Could it be possible to be inside the password box? If not, it is also fine like this.

Sanchitbajaj02 commented on src/components/pages/auth/login/index.tsx:
```diff
@@ -13,6 +13,14 @@ import { toastify } from "@/helper/toastify";
 import { loginUser } from "@/backend/auth.api";
 
 export default function LoginComponent() {
+  function showPassword(): void {
```
Is it possible to use html `event` instead of using `getElementById` for updating.

sumanjeet0012 commented:
I have added the eye button inside the password input field and removed the getElementById ,
The code is much smaller now.

I have tested the eye buttons they are working as expected.
![image](https://github.com/Sanchitbajaj02/palettegram/assets/78401409/d460fe95-6002-46b5-a101-c367c1b2b66b)

Sanchitbajaj02 commented on package-lock.json:
Please remove the package-lock.json file as it will give issue while deploying in CI

sumanjeet0012 replied:
I don't know how to remove a pushed file, So I have removed every line of code from package-lock.json and then pushed it.

Sanchitbajaj02 commented on package.json:
```diff
@@ -32,6 +32,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-feather": "^2.0.10",
+    "react-icons": "^4.12.0",
```
It would be better if you use the added react-feather icon library since it is light weight. The size of react icon is [over 70 MB](https://www.npmjs.com/package/react-icons).

sumanjeet0012 replied:
replaced react-icons from react-feather

Sanchitbajaj02 commented on src/components/pages/auth/login/index.tsx:
```diff
@@ -12,8 +12,11 @@ import { toastify } from "@/helper/toastify";
 import { loginUser } from "@/backend/auth.api";
 
+// Icons
+import { FaEye, FaEyeSlash } from "react-icons/fa";
```
Already available in the react-feather

sumanjeet0012 replied:
Now i have used react-feather

Sanchitbajaj02 commented on src/components/pages/auth/register/index.tsx:
```diff
@@ -130,21 +134,44 @@ export default function RegisterComponent() {
+                    className="absolute top-1/2 right-2 transform -translate-y-1/2 cursor-pointer"
+                    onClick={(e) => {
+                      setShowPassword(!showPassword);
+                      e.stopPropagation();
```
What is `e.stopPropagation();`? What is the use of this?

sumanjeet0012 replied:
removed this line.
